machine/v1beta1/types_powervsprovider.go

```diff
@@ -0,0 +1,156 @@
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// PowerVSMachineProviderConfig is the type that will be embedded in a Machine.Spec.ProviderSpec field
+// for a PowerVS virtual machine. It is used by the PowerVS machine actuator to create a single Machine.
+//
+// Compatibility level 2: Stable within a major release for a minimum of 9 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=2
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type PowerVSMachineProviderConfig struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// UserDataSecret contains a local reference to a secret that contains the
+	// UserData to apply to the instance.
+	// +optional
+	UserDataSecret *corev1.LocalObjectReference `json:"userDataSecret,omitempty"`
+
+	// CredentialsSecret is a reference to the secret with IBM Cloud credentials.
+	// +optional
+	CredentialsSecret *corev1.LocalObjectReference `json:"credentialsSecret,omitempty"`
+
+	// Region indicates the Power VS ServiceInstance region.
+	// +optional
+	Region string `json:"region,omitempty"`
+
+	// Zone indicates the Power VS ServiceInstance zone.
+	// +optional
+	Zone string `json:"zone,omitempty"`
+
+	// ServiceInstance is the reference to the Power VS ServiceInstance on which the machine instance will be created.
+	// +kubebuilder:validation:=Required
+	ServiceInstance PowerVSResource `json:"serviceInstance"`
+
+	// Image is the OS image to use to create the instance.
+	// +kubebuilder:validation:=Required
+	Image PowerVSResource `json:"image"`
+
+	// Network is the reference to the Network to use for this instance.
+	// +kubebuilder:validation:=Required
+	Network PowerVSResource `json:"network"`
+
+	// KeyPairName is the name of the KeyPair to use for SSH
+	// +kubebuilder:validation:=Required
+	KeyPairName string `json:"keyPairName"`
```

Comment:
ssh-key-pair name is a mandatory param to the create vm call in the powervs service, without this it actually fails. User need to create the key pair in advance and send the same to the api to use while the deploy(though it doesn't play a big role here in case of RHCOS, will ask powervs team to make this optional going forward so that we can get rid off the same)

Reply:
Can this be hidden from the end user so that we don't expose it to them? IMO it doesn't make sense to expose this to the user as it's not something that actually effects them

Reply:
> Can this be hidden from the end user so that we don't expose it to them? IMO it doesn't make sense to expose this to the user as it's not something that actually effects them

How this cab be hidden? this is a required param to the api, otherwise we have seen error while deploying the instance.

Reply:
What I meant was, could the Machine controller handle this for the user by creating a random keypair per machine (and deleting it again later), and then the user doesn't need to know about this

Reply:
Installer is already creating one per cluster, these keys are global to the account and I'm wondering how the whole system will behave when we have many more machines and users(don't want to hit with any limits here, thought not sure if there is limit).

Reply:
Is the name of that key available somewhere within the cluster? 

I have noticed that OpenStack have an SSH key pair in their API, I'm trying to gather context on that. That said, repeating past mistakes is not something we do in OpenShift

Reply:
@clnperez are we pushing ssh-key name we use using during the installer somewhere?

This means that we end up adding dependency on some of the resources like Infrastructure or any other which contains the key name, this will add the dependency on the other non-mapi resources, not sure if we really should do that!

Reply:
In general, we want to make the API our users consume as simple as possible, making them set a field that is redundant (ie this has no effect on the outcome of the VM, but they must set it to something) makes for a poor UX, so i'm a strong believer we should find some way to hide this, or make it optional from the provider side

Do we know why it is required on the provider side? Perhaps if there's some use case that we can explain then we can find a reason this needs to be exposed to the user